### PR TITLE
Display related SDG and targets on polls and processes

### PR DESCRIPTION
**Pull request by javierm**
## References

* Depends on #4324

## Objectives

* Display related SDG and targets on polls and processes

## Visual Changes

**Polls**
Captura de pantalla 2021-01-28 a las 15 35 57

Captura de pantalla 2021-01-28 a las 15 36 15

**Processes**

Captura de pantalla 2021-01-28 a las 15 36 33
Captura de pantalla 2021-01-28 a las 15 37 02

### CODE REVIEW

File: app/components/sdg/targets/plain_tag_list_component.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+              class: "more-#{i18n_namespace}",
+              title: t("sdg.#{i18n_namespace}.filter.more", count: count)
+    end
+
+    def targets
+      record.sdg_targets
+    end
+
+    def limit?
+      count_out_of_limit(targets) > 0
```
Comment (gabrielbaldao) [inline app/components/sdg/targets/plain_tag_list_component.rb:30]: In this case I like to use the `.positive?` method
https://apidock.com/rails/Numeric/positive%3F

File: app/components/sdg/goals/plain_tag_list_component.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    end
+
+    def tags
+      [*goal_tags, see_more_link(goals)]
+    end
+
+    def see_more_link(collection)
+      count = count_out_of_limit(collection)
+
+      if count > 0
```
Comment (gabrielbaldao) [inline app/components/sdg/goals/plain_tag_list_component.rb:17]: same here

File: app/components/sdg/targets/plain_tag_list_component.rb (hunk 1 above)
Comment (Senen) [reply app/components/sdg/targets/plain_tag_list_component.rb]: Thanks @gabrielbaldao, I applied your suggestion.

File: app/components/sdg/goals/plain_tag_list_component.rb [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    end
+
+    def see_more_link
+      count = count_out_of_limit(goals)
+
+      if count.positive?
+        link_to "#{count}+", polymorphic_path(record),
+                class: "more-#{i18n_namespace}",
+                title: t("sdg.#{i18n_namespace}.filter.more", count: count)
+      end
```
Comment (javierm) [inline app/components/sdg/goals/plain_tag_list_component.rb:21]:
```suggestion
      link_to(*super(goals)) if super(goals)
```

File: app/components/sdg/targets/plain_tag_list_component.rb [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    end
+
+    def see_more_link
+      count = count_out_of_limit(targets)
+
+      if count.positive?
+        link_to "#{count}+", polymorphic_path(record),
+                class: "more-#{i18n_namespace}",
+                title: t("sdg.#{i18n_namespace}.filter.more", count: count)
+      end
```
Comment (javierm) [inline app/components/sdg/targets/plain_tag_list_component.rb:25]:
```suggestion
      link_to(*super(targets)) if super(targets)
```

File: app/assets/stylesheets/legislation.scss [hunk 5]
```diff
@@ -69,10 +69,14 @@
 }
 
 .legislation-text {
-  margin-bottom: 1rem;
```
Comment (javierm) [inline app/assets/stylesheets/legislation.scss:72]: I made a mistake here. This line should be kept as well; I removed it thinking it applied to the `.legislation-text` element which is removed in this pull request, but it applies to the element which it **not** removed as well.

Thanks @taitus for noticing.

**Review by javierm — COMMENTED** [review]
Here's the initial review. Let me know what you think!

File: app/assets/stylesheets/legislation.scss (hunk 5 above)
Comment (Senen) [reply app/assets/stylesheets/legislation.scss:72]: Done!

**Review by taitus — APPROVED** [review]

File: spec/components/sdg/goals/plain_tag_list_component_spec.rb [hunk 6]
```diff
@@ -0,0 +1,56 @@
+require "rails_helper"
+
+describe SDG::Goals::PlainTagListComponent, type: :component do
+  let(:debate) { create(:debate, sdg_goals: [SDG::Goal[1], SDG::Goal[3]]) }
+  let(:component) { SDG::Goals::TagListComponent.new(debate) }
```
Comment (javierm) [inline spec/components/sdg/goals/plain_tag_list_component_spec.rb:5]: These tests aren't testing the `PlainTagListComponent` but ghe `TagListComponent`.
```suggestion
  let(:component) { SDG::Goals::PlainTagListComponent.new(debate) }
```

File: spec/system/legislation/processes_spec.rb [hunk 7]
```diff
@@ -127,6 +127,19 @@
         expect(page).to have_content("past published")
       end
     end
+
+    scenario "Show SDG tags when feature is enabled" do
```
Comment (javierm) [inline spec/system/legislation/processes_spec.rb:131]:
```suggestion
    scenario "Show SDG tags when feature is enabled", :js do
```

File: spec/system/legislation/processes_spec.rb [hunk 8]
```diff
@@ -127,6 +127,19 @@
...
+      Setting["feature.sdg"] = true
+      Setting["sdg.process.legislation"] = true
+
+      create(:legislation_process, sdg_goals: [SDG::Goal[1]],
+                                   sdg_targets: [SDG::Target["1.1"]])
+
+      visit legislation_processes_path
+
+      expect(page).to have_selector(".sdg-goal-tag-list")
+      expect(page).to have_selector(".sdg-target-tag-list")
```
Comment (javierm) [inline spec/system/legislation/processes_spec.rb:141]: How about testing it from the user's point of view?

```suggestion
      expect(page).to have_selector "img[alt='1. No Poverty']"
      expect(page).to have_content "target 1.1"
```

File: spec/system/legislation/processes_spec.rb [hunk 9]
```diff
@@ -210,6 +223,19 @@
...
+        Setting["feature.sdg"] = true
+        Setting["sdg.process.legislation"] = true
+
+        process = create(:legislation_process, sdg_goals: [SDG::Goal[1]],
+                                               sdg_targets: [SDG::Target["1.1"]])
+
+        visit legislation_process_path(process)
+
+        expect(page).to have_selector(".sdg-goal-tag-list")
+        expect(page).to have_selector(".sdg-target-tag-list")
```
Comment (javierm) [inline spec/system/legislation/processes_spec.rb:237]: How about testing it from the user's point of view?

```suggestion
      expect(page).to have_selector "img[alt='1. No Poverty']"
      expect(page).to have_content "target 1.1"
```

File: spec/system/polls/polls_spec.rb [hunk 10]
```diff
@@ -129,6 +129,19 @@
...
+      Setting["feature.sdg"] = true
+      Setting["sdg.process.polls"] = true
+
+      create(:poll, sdg_goals: [SDG::Goal[1]],
+                    sdg_targets: [SDG::Target["1.1"]])
+
+      visit polls_path
+
+      expect(page).to have_selector(".sdg-goal-tag-list")
+      expect(page).to have_selector(".sdg-target-tag-list")
```
Comment (javierm) [inline spec/system/polls/polls_spec.rb:143]: How about testing it from the user's point of view?

```suggestion
      expect(page).to have_selector "img[alt='1. No Poverty']"
      expect(page).to have_content "target 1.1"
```

File: spec/system/polls/polls_spec.rb [hunk 11]
```diff
@@ -401,6 +414,19 @@
...
+      Setting["feature.sdg"] = true
+      Setting["sdg.process.polls"] = true
+
+      poll = create(:poll, sdg_goals: [SDG::Goal[1]],
+                           sdg_targets: [SDG::Target["1.1"]])
+
+      visit poll_path(poll)
+
+      expect(page).to have_selector(".sdg-goal-tag-list")
+      expect(page).to have_selector(".sdg-target-tag-list")
```
Comment (javierm) [inline spec/system/polls/polls_spec.rb:428]: How about testing it from the user's point of view?

```suggestion
      expect(page).to have_selector "img[alt='1. No Poverty']"
      expect(page).to have_content "target 1.1"
```

File: spec/system/legislation/processes_spec.rb [hunk 12]
```diff
@@ -210,6 +223,19 @@
 
         expect(page).to have_current_path new_legislation_process_proposal_path(process)
       end
+
+      scenario "Show SDG tags when feature is enabled" do
```
Comment (javierm) [inline spec/system/legislation/processes_spec.rb:227]:
```suggestion
      scenario "Show SDG tags when feature is enabled", :js do
```

File: spec/system/polls/polls_spec.rb [hunk 13]
```diff
@@ -129,6 +129,19 @@
 
       expect(page).to have_link("Poll with results", href: results_poll_path(poll.slug))
     end
+
+    scenario "Shows SDG tags when feature is enabled" do
```
Comment (javierm) [inline spec/system/polls/polls_spec.rb:133]:
```suggestion
    scenario "Shows SDG tags when feature is enabled" , :js do
```

File: spec/system/polls/polls_spec.rb [hunk 14]
```diff
@@ -401,6 +414,19 @@
         expect(page).to have_link("Yes")
       end
     end
+
+    scenario "Shows SDG tags when feature is enabled" do
```
Comment (javierm) [inline spec/system/polls/polls_spec.rb:418]:
```suggestion
    scenario "Shows SDG tags when feature is enabled", :js do
```

**Pull request merged**